test/benchmark_utils/test_benchmark_utils.py

```diff
@@ -0,0 +1,907 @@
+import json
+import os
+import re
+import textwrap
+from typing import Any, List, Tuple
+import unittest
+
+import torch
+import torch.utils.benchmark as benchmark_utils
+from torch.testing._internal.common_utils import TestCase, run_tests, IS_WINDOWS, slowTest
+from torch.testing._internal import expecttest
+import numpy as np
+
+
+CALLGRIND_ARTIFACTS: str = os.path.join(
+    os.path.split(os.path.abspath(__file__))[0],
+    "callgrind_artifacts.json"
+)
+
+
+def generate_callgrind_artifacts() -> None:
```

Comment:
This function isn't called by anything, is that right? Should have a comment explaining why...

Reply:
Done.